### PR TITLE
feat(LT-3316): add api for rfq monitoring with taking data from db

### PR DESCRIPTION
**Pull request by IuriiAksenov**

### CODE REVIEW

File: src/MarginTrading.SqlRepositories/Repositories/OperationExecutionInfoRepository.cs [hunk 1]
```diff
@@ -91,13 +93,49 @@ public class OperationExecutionInfoRepository : IOperationExecutionInfoRepositor
...
+                            + (from == null ? "" : " AND [LastModified] >= @from")
+                            + (to == null ? "" : " AND [LastModified] < @to");
+
+            var whereJson = (string.IsNullOrWhiteSpace(instrumentId) ? "" : " AND JSON_VALUE([Data], '$.Instrument') = @instrumentId")
+                          + (string.IsNullOrWhiteSpace(accountId) ? "" : " AND JSON_VALUE([Data], '$.AccountId') = @accountId")
+                          + (states == null || states.Count == 0 ? "" : $" AND JSON_VALUE([Data], '$.State') in ({string.Join(",", states.Select(x => $"'{x}'"))})");
+
+            var whereClause = whereRfq + whereFields + whereJson;
+
+            const int MaxResults = 1000;
```
Comment (tarurar) [inline src/MarginTrading.SqlRepositories/Repositories/OperationExecutionInfoRepository.cs:112]: I think 1000 is too much. Let's make it 100.

**Review by tarurar — CHANGES_REQUESTED** [review]

File: src/MarginTrading.Backend/Controllers/RfqController.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                Id = operation.Id,
+                InstrumentId = operation.Data.Instrument,
+                PositionIds = operation.Data.PositionIds,
+                Volume = operation.Data.Volume,
+                Price = operation.Data.Price,
+                ExternalProviderId = operation.Data.ExternalProviderId,
+                AccountId = operation.Data.AccountId,
+                CausationOperationId = operation.Data.CausationOperationId,
+                CreatedBy = string.IsNullOrEmpty(operation.Data.AdditionalInfo) ? null : JsonConvert.DeserializeObject<AdditionalInfo>(operation.Data.AdditionalInfo).CreatedBy,
```
[the remark this answers is not in the document]
Comment (IuriiAksenov) [reply src/MarginTrading.Backend/Controllers/RfqController.cs:56]: IsOnBehalf duplicates OriginatorType, when IsOnBehalf true OriginatorType has also OnBehalf value

File: src/MarginTrading.Backend/Controllers/RfqController.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        public RfqController(IOperationExecutionInfoRepository operationExecutionInfoRepository)
+        {
+            _operationExecutionInfoRepository = operationExecutionInfoRepository;
+        }
+
+        [HttpGet]
+        public async Task<PaginatedResponseContract<RfqContract>> GetAsync([CanBeNull, FromQuery] GetRfqRequest getRfqRequest, [FromQuery] int skip = 0, [FromQuery] int take = 20)
+        {
+            var states = getRfqRequest.States?.Select(x => (SpecialLiquidationOperationState)x).ToList();
```
[the remark this answers is not in the document]
Comment (IuriiAksenov) [reply src/MarginTrading.Backend/Controllers/RfqController.cs:37]: Added

File: src/MarginTrading.SqlRepositories/Repositories/OperationExecutionInfoRepository.cs (hunk 1 above)
Comment (IuriiAksenov) [reply src/MarginTrading.SqlRepositories/Repositories/OperationExecutionInfoRepository.cs]: Fixed

**Review by tarurar — APPROVED** [review]

**Review by gponomarev-lykke — APPROVED** [review]

**Pull request merged**